src/peergos/shared/user/ServerConversation.java

```diff
@@ -0,0 +1,49 @@
+package peergos.shared.user;
+
+import jsinterop.annotations.*;
+
+import java.util.*;
+
+public class ServerConversation implements Comparable<ServerConversation> {
+
+    public final long latestEpochMillis;
+    @JsProperty
+    public final List<ServerMessage> messages;
+    public final boolean isDismissed;
+
+    public ServerConversation(List<ServerMessage> messages) {
+        this.messages = messages;
+        long latest = 0;
+        for (ServerMessage message : messages) {
+            if (message.sentEpochMillis > latest)
+                latest = message.sentEpochMillis;
+        }
+        ServerMessage lastMessage = messages.get(messages.size() - 1);
+        this.isDismissed = lastMessage.isDismissed || lastMessage.type == ServerMessage.Type.FromUser;
```

Comment:
The last part of this statement is surprising (but i can see why you did it).

Reply:
Isn't it just the logic you asked for?

Reply:
Yes. It is the correct logic to filter for display. Using the logic to set a variable called isDismissed is what i question.

Reply:
Oh sure. I can rename it. isHidden?

Reply:
IsDisplayable maybe?

Reply:
done